src/performance.js

```diff
@@ -60,6 +68,49 @@ export class Performance {
     this.events_ = [];
   }
 
+  /**
+   * Listens to viewer and resource events.
+   */
+  coreServicesAvailable() {
+    this.viewer_ = viewerFor(this.win);
+    this.resources_ = resourcesFor(this.win);
+
+    this.viewer_.onVisibilityChanged(this.flush.bind(this));
+
+    let timeReqToBeVisible = 0;
+
+    // If doc didnt start in prerender mode, then no need to
+    // listen for when the first resource is laid out.
+    if (this.viewer_.didStartInPrerender()) {
+      this.resources_.whenFirstResourceLaidOut().then(() => {
+        if (this.viewer_.hasBeenVisible()) {
+          // Prerender Completed, and user has requested
+          // doc to be visible prior to prerender completion.
+          // User might have been seeing the first set of resources
+          // loading.
+          this.tick('pc', null, timer.now() - timeReqToBeVisible);
```

Comment:
@dvoytenko after collecting the elements, what would be the indicator that the element is done? (tracking it and marking it as done in layoutComplete or is there an easier way to hook into it?)